sig/datadog/core/utils/network.rbs

```diff
@@ -0,0 +1,23 @@
+module Datadog
+  module Core
+    module Utils
+      module Network
+        DEFAULT_IP_HEADERS_NAMES: ::Array[::String]
+
+        def self.raw_ip_from_request_headers: (Datadog::Core::HeaderCollection headers, ?::Array[::String] ip_headers_to_check) -> ::String?
+        def self.parse_ip: (::String ip) -> untyped?
+
+        private
+
+        def self.ip_header: (Datadog::Core::HeaderCollection headers, ::Array[::String] ip_headers_to_check) -> untyped?
+        def self.strip_zone_specifier: (::String ipv6) -> ::String
+        def self.strip_ipv6_port: (::String ip) -> ::String
+        def self.strip_ipv4_port: (::String ip) -> ::String
+        def self.global_ip?: (untyped parsed_ip) -> bool
+        def self.private?: (untyped ip) -> bool
+        def self.link_local?: (untyped ip) -> bool
+        def self.loopback?: (untyped ip) -> bool
```

Comment:
Minor: I'm curious, why are all the `IPaddr`s here showing up as `untyped` -- is there a limitation to using them here?

Reply:
I haven't figured yet, how to include the standard library `rbs` files in the project. I decided to leave this one outside of this PR to reduce its size of it.

I will tackle that in a separate PR